packages/solana/src/connectors/WalletConnectConnector.ts

```diff
@@ -0,0 +1,254 @@
+    }
+    const transaction = transactionParam as Transaction
+    const transactionParams = {
+      feePayer: transaction.feePayer?.toBase58() ?? '',
+      instructions: transaction.instructions.map(instruction => ({
+        data: base58.encode(instruction.data),
+        keys: instruction.keys.map(key => ({
+          isWritable: key.isWritable,
+          isSigner: key.isSigner,
+          pubkey: key.pubkey.toBase58()
+        })),
+        programId: instruction.programId.toBase58()
+      })),
+      recentBlockhash: transaction.recentBlockhash ?? ''
+    }
+
+    const res = await this.request('solana_signTransaction', transactionParams)
+    transaction.addSignature(
+      new PublicKey(SolStoreUtil.state.address ?? ''),
+      Buffer.from(base58.decode(res.signature))
+    )
+
+    const validSig = transaction.verifySignatures()
+
+    if (!validSig) {
+      throw new Error('Signature invalid.')
+    }
+
+    return { signatures: [{ signature: base58.encode(transaction.serialize()) }] }
+  }
+
+  public async sendTransaction(transactionParam: Transaction | VersionedTransaction) {
+    const encodedTransaction = (await this.signTransaction(transactionParam)) as {
+      signatures: {
+        signature: string
+      }[]
+    }
+    const signedTransaction = base58.decode(encodedTransaction.signatures[0]?.signature ?? '')
+    await SolStoreUtil.state.connection?.sendRawTransaction(signedTransaction)
+
+    return base58.encode(signedTransaction)
+  }
+
+  /**
+   * Connect to user's wallet.
+   *
+   * If `WalletConnectConnector` was configured with `qrcode = true`, this will
+   * open a QRCodeModal, where the user will scan the qrcode and then this
+   * function will resolve/return the address of the wallet.
+   *
+   * If `qrcode = false`, this will return the pairing URI used to generate the
+   * QRCode.
+   */
+  public generateNamespaces(chainId: string) {
+    const rpcMap = {
+      [`solana:${solana.chainId}`]: solana.rpcUrl,
+      [`solana:${solanaTestnet.chainId}`]: solanaTestnet.rpcUrl,
+      [`solana:${solanaDevnet.chainId}`]: solanaDevnet.rpcUrl
+    }
+    const chainsNamespaces = [`solana:${chainId}`]
+    const rpc = {
+      [chainId]: rpcMap[`solana:${chainId}`] ?? ''
+    }
+
+    return {
+      solana: {
+        chains: [...chainsNamespaces],
+        methods: ['solana_signMessage', 'solana_signTransaction'],
+        events: [],
+        rpcMap: rpc
+      }
+    }
+  }
+
+  public async connect(useURI?: boolean) {
+    const solanaNamespace = this.generateNamespaces(SolStoreUtil.state.currentChain?.chainId ?? '')
+
+    const provider = await UniversalProviderFactory.getProvider()
+
+    return new Promise<string>((resolve, reject) => {
+      provider.on('display_uri', (uri: string) => {
+        if (!(this.qrcode && !useURI)) {
+          resolve(uri)
+        }
+      })
+      provider
+        .connect({
+          pairingTopic: undefined,
+          namespaces: solanaNamespace
```

Comment:
```suggestion
          optionalNamespaces: solanaNamespace
```
let's use optionalNamespaces instead, should be the default